### PR TITLE
Merge to from Develoment to main branch

### PR DESCRIPTION
**Pull request by Caren-Koroeny**
<h1>Hello :wave:  Code Reviwer(s)</h1>

> In this Branch we implemented the Following for our Ruby capstone project.

- [ ] Created a main.rb file that will serve as the console app entry point.

- [ ] The console app, at the start, does the following:

:white_circle: Presents the user with a list of options to perform.
:white_circle: Let users choose an alternative.
:white_circle: If you need it, please ask for the parameters for the option.
:white_circle: Have a way to quit the app.

The following options were made available:

:white_check_mark: List all books
:white_check_mark: List all music albums
 :white_check_mark: List all movies
:white_check_mark: List of games
 :white_check_mark: List all genres (e.g 'Comedy', 'Thriller')
:white_check_mark: List all labels (e.g. 'Gift', 'New')
:white_check_mark: List all authors (e.g. 'Stephen King')
 :white_check_mark: List all sources (e.g. 'From a friend, 'Online shop')
:white_check_mark: Add a book
 :white_check_mark: Add a music album
:white_check_mark: Add a movie
 :white_check_mark: Add a game

:heavy_check_mark: All data was preserved by saving collections in .json files.

:bug: Run Rspec test to test the functionalities of the methods created.

- [ ] Created a schema.sql file with tables that will be analogical to the structure of the classes in your app.

<h1> :brain: Knowledge We acquired from the whole Project :tanabata_tree: </h1>

:white_check_mark: Insert and query data in SQL.

:white_check_mark: Use primary crucial & foreign key mechanisms for joining tables.

:white_check_mark: Understand the different types of relationships between tables.

:white_check_mark: Query multiple tables.

:white_check_mark: Prepare complex queries that answer analytical questions.

:white_check_mark: Run a program using the command line.

:white_check_mark: Use Ruby syntax for basic programming operations.

:white_check_mark: Apply Ruby best practices and language style guides in code.

:white_check_mark: Describe the SOLID principles of OOP.

:white_check_mark: Implement classes and objects in Ruby.

:white_check_mark: Understand the four main principles of OOP.

:white_check_mark: Implement encapsulation and inheritance with Ruby.

:white_check_mark: Create a UML class diagram.

:white_check_mark: Explain the difference between associations, aggregations, and composition in OOP.

:white_check_mark: Set up associations between classes and objects.

:white_check_mark: Store data in files.

:white_check_mark: Build interactive console apps.

Recorded a 5mins [video](https://drive.google.com/file/d/1twF_CErAK3xzdBDKWHj9aPSwB7a5Ius6/view?usp=sharing) showing the code and how it works

<h2> Kindly Approve 👍🏼 :100: </h2>

### CODE REVIEW

File: classes/item.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Item
+  attr_accessor :genre, :author, :label, :archived, :publish_date
```
Comment (mikethreels) [inline classes/item.rb:2]: - [ ] in the Item class you should add the "has one" from the "one-to-many" relationship for Author, Label and genre, which means you have to add a method for each class which looks something like below:
```
    def genre=(genre)
      @genre = genre
      genre.items.push(self) unless genre.items.include?(self)
    end

```

File: classes/book.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  def initialize(publisher, cover_state, publish_date, author, title)
+    super(publish_date)
+    @author = author
+    @title = title
+    @publisher = publisher
+    @cover_state = cover_state
+  end
+
+  def can_be_archived?
+    super() && (@cover_state == 'bad' || can_be_archived?)
```
Comment (mikethreels) [inline classes/book.rb:15]: - [ ] you are already calling super, therefore no need to call the method again.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Great job setting up the application.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: :raising_hand: @Caren-Koroeny, @mikethreels 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 

:tada: :champagne:  :tada: :champagne: 

## Highlights :green_circle: 

- PR is professional :+1: 
- Console app works without errors :+1: 
- Good commit history :+1: 
- Descriptive README.md file :+1: 
- Linter checks passed :heavy_check_mark: 

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**